gateways/community/aws-sqs/validate.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 BlackRock, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws_sqs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/argoproj/argo-events/gateways"
+	gwcommon "github.com/argoproj/argo-events/gateways/common"
+)
+
+// ValidateEventSource validates gateway event source
+func (ese *SQSEventSourceExecutor) ValidateEventSource(ctx context.Context, es *gateways.EventSource) (*gateways.ValidEventSource, error) {
+	return gwcommon.ValidateGatewayEventSource(es.Data, parseEventSource, validateSQSConfig)
+}
+
+func validateSQSConfig(config interface{}) error {
```

Comment:
The more I think about these config maps.. the more it possibly makes sense to instantiate a new EventSource CRD so they these can benefit from admission webhook validation as well as clean API type safety. We can discuss this offline as it could be a v2 thing...

Reply:
agreed.